tools/validators/instance_validator/validate/entity_instance.py

```diff
@@ -60,6 +62,69 @@ def _ValidateType(self):
 
     return True
 
+  def _ValidateTranslation(self):
+    """Uses information from the generated ontology universe to validate
+    an entity's translation if it exists.
+
+    Returns:
+      Returns boolean for validity of entity translation, defaults to True if
+      translation is not specified.
+    """
+    if self.translation_key not in self.entity.keys():
+      return True
+
+    translation_body = self.entity[self.translation_key]
+    if isinstance(translation_body.data, str):
+      if translation_body.data == self.translation_compliant:
+        return True
+      else:
+        print('Invalid translation compliance string', translation_body.data)
+        return False
+
+    # iterate through each translation device key and determine its form
+    for device_name in translation_body.keys():
+      # check if keys are UDMI compliant
+      if isinstance(translation_body[device_name].data, str):
+        continue
+
+      device_map = translation_body[device_name]
+      valid_units = self.universe.unit_universe.GetUnitsMap('').keys()
+      valid_states = self.universe.state_universe.GetStatesMap('').keys()
+
+      # three remaining possibilities for translation format
+      if 'unit_values' in device_map.keys():
+        unit_values_map = device_map['unit_values']
+        for unit in unit_values_map.keys():
+          if unit not in valid_units:
+            print('Invalid translation unit', unit)
+            return False
+      elif 'states' in device_map.keys():
+        states_map = device_map['states']
+        for state in states_map.keys():
+          if state not in valid_states:
+            print('Invalid translation state', state)
+            return False
+      elif 'units' in device_map.keys():
+        # check that the unit map has keys named `keys`, `values`
+        units_map = device_map['units']
+        if 'key' not in units_map.keys():
```

Comment:
define `key` as a constant and the same for each `string value` used

Reply:
Done